<launch>
  <!-- Humanoid LoadMap -->
  <include file="$(find humanoid_loadmap)/launch/loadmap.launch"/>
  <!-- Humanoid Interface -->
  <arg name="device_name"      default="/dev/ttyUSB0"/>
  <arg name="baud_rate"        default="1000000"/>
  <arg name="protocol_version" default="2.0"/>
  <arg name="toRobot"          default="true"/>
  <arg name="toGazebo"         default="false"/>
  <arg name="dt"               default="0.07"/>

  <param name="humanoid_interface/device_name"      value="$(arg device_name)"/>
  <param name="humanoid_interface/baud_rate"        value="$(arg baud_rate)"/>
  <param name="humanoid_interface/protocol_version" value="$(arg protocol_version)"/>
  <param name="humanoid_interface/toRobot"          value="$(arg toRobot)"/>
  <param name="humanoid_interface/toGazebo"         value="$(arg toGazebo)"/>
  <param name="humanoid_interface/dt"               value="$(arg dt)"/>


  <node name="humanoid_interface" pkg="humanoid_interface" type="humanoid_interface_node" output="screen" >   </node>

   <!-- MotorSet -->
  <param name="motor_set_control/toRobot"          value="$(arg toRobot)"/>
  <param name="motor_set_control/toGazebo"         value="$(arg toGazebo)"/>
  <param name="motor_set_control/dt"               value="$(arg dt)"/>

  <node name="motor_set_control" pkg="motor_set_control" type="motorset_node" output="screen" >   </node>

   <!-- HumanoidModel -->
  <rosparam file="$(find humanoid_model)/config/robotParams.yaml" command="load"/>
  <param name="humanoid_model/movcreator/on"              value="true"/>
  <param name="humanoid_model/movcreator/toJointState"    value="false"/>
  <param name="humanoid_model/movcreator/toMotorState"    value="false"/>
  <param name="humanoid_model/walk/calcIK"                value="false"/>
  <node name="humanoid_model" pkg="humanoid_model" type="humanoidmodel_node" output="screen" >   </node>
   <!-- Humanoid Config -->
  <node name="humanoid_config" pkg="rqt_reconfigure" type="rqt_reconfigure" output="screen" >   </node>


   <!-- MovCreatorQt -->
  <node name="movcreator_qt" pkg="movcreator_qt" type="movcreator_qt" output="screen" >   </node>


</launch>
